### PR TITLE
Updated Elasticsearch upstream to 1.5.2

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: sql/src/test/java/io/crate/metadata/doc/array/ArrayMapperMetaMigrationTest.java [hunk 1]
```diff
@@ -46,6 +48,10 @@
 
 public class ArrayMapperMetaMigrationTest extends CrateUnitTest {
 
+    static {
+        Loggers.getLogger(ArrayMapperMetaMigration.class).setLevel("TRACE");
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/metadata/doc/array/ArrayMapperMetaMigrationTest.java]: can you remove that?

File: blob/src/main/java/org/elasticsearch/indices/recovery/BlobRecoverySource.java [hunk 2]
```diff
@@ -496,5 +159,51 @@ public void messageReceived(final StartRecoveryRequest request, final TransportC
             channel.sendResponse(response);
         }
     }
+
+
+    private static final class OngoingRecoveres {
```
Comment (mfelsche) [inline blob/src/main/java/org/elasticsearch/indices/recovery/BlobRecoverySource.java]: typo

File: es/build.gradle [hunk 3]
```diff
@@ -36,32 +38,35 @@ dependencies {
...
+    compile 'io.netty:netty:3.10.0.Final'
+    compile 'joda-time:joda-time:2.7'
+    compile 'log4j:apache-log4j-extras:1.2.17'
     compile 'log4j:log4j:1.2.17'
     compile 'net.java.dev.jna:jna:4.1.0'
     compile 'org.antlr:antlr-runtime:3.5'
     compile 'org.apache.commons:commons-lang3:3.3.2'
-    compile 'org.codehaus.groovy:groovy-all:2.3.2'
+    compile 'org.codehaus.groovy:groovy-all:2.4.0'
+    compile 'org.fusesource:sigar:1.6.4'
```
Comment (mfelsche) [inline es/build.gradle]: please exclude

**Comment by mfelsche** [comment]
fix thread leak filter, otherwise ack

**Comment by seut** [comment]
ack (rebase ;)

**Pull request merged**